addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Device.java

```diff
@@ -0,0 +1,352 @@
+     */
+    public void resetAllUpdates(HashMap<String, KM200CommObject> actTreeMap) {
+        if (actTreeMap != null) {
+            for (KM200CommObject stmObject : actTreeMap.values()) {
+                if (stmObject != null) {
+                    stmObject.setUpdated(false);
+                    resetAllUpdates(stmObject.serviceTreeMap);
+                }
+            }
+        }
+    }
+
+    /**
+     * This function checks whether a service is available
+     *
+     * @param service
+     */
+    public Boolean containsService(String service) {
+        String[] servicePath = service.split("/");
+        KM200CommObject object = null;
+        int len = servicePath.length;
+        if (len == 0) {
+            return false;
+        }
+        if (!serviceTreeMap.containsKey(servicePath[1])) {
+            return false;
+        } else {
+            if (len == 2) {
+                return true;
+            }
+            object = serviceTreeMap.get(servicePath[1]);
+        }
+        for (int i = 2; i < len; i++) {
+            if (object.serviceTreeMap.containsKey(servicePath[i])) {
+                object = object.serviceTreeMap.get(servicePath[i]);
+                continue;
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This function return the KM200CommObject of a service
+     *
+     * @param service
+     */
+    public KM200CommObject getServiceObject(String service) {
+        String[] servicePath = service.split("/");
+        KM200CommObject object = null;
+        int len = servicePath.length;
+        if (len == 0) {
+            return null;
+        }
+        if (!serviceTreeMap.containsKey(servicePath[1])) {
+            return null;
+        } else {
+            object = serviceTreeMap.get(servicePath[1]);
+            if (len == 2) {
+                return object;
+            }
+        }
+        for (int i = 2; i < len; i++) {
+            if (object.serviceTreeMap.containsKey(servicePath[i])) {
+                object = object.serviceTreeMap.get(servicePath[i]);
+                continue;
+            } else {
+                return null;
+            }
+        }
+        return object;
+    }
+
+    // setter
+    public void setIP4Address(String ip) {
+        ip4Address = ip;
+    }
+
+    public void setGatewayPassword(String password) {
+        gatewayPassword = password;
+        recreateKeys();
+    }
+
+    public void setPrivatePassword(String password) {
+        privatePassword = password;
+        recreateKeys();
+    }
+
+    public void setMD5Salt(String salt) {
+        MD5Salt = DatatypeConverter.parseHexBinary(salt);
+        recreateKeys();
+    }
+
+    public void setCryptKeyPriv(String key) {
+        cryptKeyPriv = DatatypeConverter.parseHexBinary(key);
+    }
+
+    public void setCharSet(String charset) {
+        charSet = charset;
+    }
+
+    public void setInited(Boolean Init) {
```

Comment:
`Init` does not match the naming guidelines Why `Boolean `(the object one) and not just `boolean`

Reply:
The reason was to use objects only. But I will change it. OK.